esp/esp/program/modules/handlers/studentregphasezero.py

```diff
@@ -161,6 +122,57 @@ def studentregphasezero(self, request, tl, one, two, module, extra, prog):
 
     @aux_call
     @needs_student
+    def joingroup(self, request, tl, one, two, module, extra, prog, newclass = None):
+        context = {}
+        context['program'] = prog
+        context['one'] = one
+        context['two'] = two
+        user = request.user
+        in_lottery = PhaseZeroRecord.objects.filter(user=user, program=prog).exists()
+        
+        join_error = False
+        if len(request.POST.get('student_selected', '').strip()) == 0:
+            join_error = 'Error - You must select a student\'s username.'
+
+        else:
+            join_user = request.POST['student_selected']
+            try:
+                group = PhaseZeroRecord.objects.get(user=join_user, program=prog)
+            except DoesNotExist:
```

Comment:
Looks like you are correct! https://docs.djangoproject.com/en/1.11/ref/models/querysets/#django.db.models.query.QuerySet.get
You can also just use a generic `ObjectDoesNotExist` exception, but that needs to be imported.